src/run-r.ts

```diff
@@ -0,0 +1,105 @@
+import * as vscode from "vscode";
+import { openBrowser, openBrowserWhenReady } from "./net-utils";
+import {
+  envVarsForShell as envVarsForTerminal,
+  escapeCommandForTerminal,
+} from "./shell-utils";
+import { getAppPort } from "./port-settings";
+
+export async function runApp(): Promise<void> {
+  const path = getActiveEditorFile();
+  if (!path) {
+    return;
+  }
+
+  const port = await getAppPort("run", "r");
+  // TODO: Is this needed for Shiny for R too?
+  // const autoreloadPort = await getAutoreloadPort("run");
+
+  const terminal = await createTerminalAndCloseOthersWithSameName({
+    name: "Shiny",
+    env: {
+      // We save this here so escapeCommandForTerminal knows what shell
+      // semantics to use when escaping arguments. A bit magical, but oh well.
+      ...envVarsForTerminal(),
+    },
+  });
+
+  const useDevmode = vscode.workspace
+    .getConfiguration("shiny.r")
+    .get("devmode");
+
+  const devOrReload = useDevmode
+    ? "shiny::devmode()"
+    : "options(shiny.autoreload = TRUE)";
+
+  const runApp = `${devOrReload}; shiny::runApp("${path}", port=${port}, launch.browser=FALSE)`;
```

Comment:
`${path}` needs escaping here. Not only for edge cases but on Windows the path will contain backslashes (I assume).

Reply:
IIUC `escapeCommandForTerminal()` handles this, It's pretty aggressive and has worked in my testing. In fact, the R code in `runApp` isn't inside quotes because it's handled correctly by `escapeCommandForTerminal()` (on my system at least, I may be missing something about how that function works).

Reply:
There are actually two issues happening here:

1. I've just used `Rscript` as the command which likely won't work on Windows, where we might need the full path to the Rscript binary, e.g. `"C:\Program Files\R\R-3.6.0\bin\Rscript.exe"`.

2. Typically, I'd use `Rscript -e 'shiny::runApp("app.R")'` in my terminal, but when I used this formulation `escapeCommandForTerminal()` broke the command by escaping `"` within the `''` quotes.

Given the above, I've refactored the `rRunApp()` command to use a packaged `runShinyApp.R` script that takes commandline arguments for `path`, `port` and `--devmode`. 

That means that the escaping now rows as expected, but leaves us with the problem of figuring out the full path to `Rscript.exe` on Windows. Do you have any ideas about how to approach that problem?

Reply:
This might not be as bad as I was thinking. I paired with @andrie this morning, who uses [rig](https://github.com/r-lib/rig) to install R on Windows, and in Powershell 7 and `cmd.exe` `Rscript` can be called directly. I'm pretty sure this is because `rig` adds `.bat` files to create an alias to run `Rscript`.

Reply:
Did you refactor to use `runShinyApp.R` only because of the `escapeCommandForTerminal()` behavior? We can definitely fix that.

Reply:
I don't think we can count on rig, this is my first time even hearing of it. You can see how the R extension does it here (and RStudio does something similar):

https://github.com/REditorSupport/vscode-R/blob/d582b9184838a32352f624e451d4c1856f347b9c/src/util.ts#L98  
https://github.com/REditorSupport/vscode-R/blob/d582b9184838a32352f624e451d4c1856f347b9c/src/util.ts#L63

Reply:
@jcheng5 I've implemented this using the vscode-R extension as inspiration. I also tested locally on a Windows VM

Reply:
> Did you refactor to use `runShinyApp.R` only because of the `escapeCommandForTerminal()` behavior? We can definitely fix that.

It might have started out this way, but we also run `shiny::devmode()` or set `options(shiny.autoreload)` before running the app. With that and the port-setting logic, I found it a lot easier to use a separate R script than to try to build the R code in strings. 

I did add some logic to echo the `shiny::runApp()` command so users could copy and paste the command into an R console.